sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -81,6 +89,109 @@ def __init__(
             )
         self.job_kind = job_kind
 
+    def train(
+        self,
+        name: str = None,
+        namespace: str = None,
+        num_workers: int = 1,
+        num_procs_per_worker: int = 1,
+        pvc: Dict[Literal["name", "claimName"], str] = None,
+        model_params: HuggingFaceModelParams = None,
+        dataset_params: S3DatasetParams = None,
+        parameters: HuggingFaceTrainParams = None,
+        resources_per_worker: Dict[Literal["gpu", "cpu", "memory"], any] = None,
+    ):
+        """
+        Higher level train api
+        """
+        if (
+            not name
+            or not namespace
+            or not pvc
+            or not model_params
+            or not dataset_params
+            or not parameters
+            or not resources_per_worker
+        ):
+            raise ValueError("One of the required parameters is None")
+
+        if num_procs_per_worker > resources_per_worker["gpu"]:
+            raise ValueError("Insufficient gpu resources allocated to the container.")
+
+        if isinstance(model_params, HuggingFaceModelParams):
+            mp = "hf"
+
+        if isinstance(dataset_params, S3DatasetParams):
+            dp = "s3"
+
+        # create init container spec
+        init_container_spec = utils.get_container_spec(
+            name=constants.JOB_PARAMETERS[constants.PYTORCHJOB_KIND]["init_container"],
+            image=constants.JOB_PARAMETERS[constants.PYTORCHJOB_KIND][
+                "init_container_image"
+            ],
+            args=[
+                "--model_provider",
+                mp,
+                "--model_provider_args",
```

Comment:
Can we keep naming consistent pls ?
E.g. name it `model_provider_parameters` and `model_parameters` or `model_provider_args` and `model_args`.